core/src/test/java/org/jdbi/v3/core/statement/TestScript.java

```diff
@@ -98,9 +101,44 @@ public void testScriptAsSetOfSeparateStatements() {
     @Test
     public void testPostgresJsonExtractTextOperator() {
         Handle h = pgExtension.openHandle();
-        Script script = h.createScript(getResourceOnClasspath("script/postgres-json-operator.sql"));
+        Script script = h.createScript(getClasspathSqlLocator().getResource("script/postgres-json-operator.sql"));
         script.execute();
 
         assertThat(h.select("select * from something").mapToMap()).hasSize(1);
     }
+
+    /**
+     * <p>
+     *   Test for correct handling of semicolons in sql containing begin/end blocks.
+     * </p>
+     * <p>
+     *   Class {@link Script} splits sql scripts into lists of statements by semicolon ({@code ;}) and then batch-executes them.<br>
+     *   Statements may contain {@code BEGIN...END} blocks containing subordinated statements (also ending in semicolons).<br>
+     *   Only semicolons on the highest level (i.e. outside any block) actually signal the end of an sql statement.
+     * </p>
+     * @author Markus Spann
+     */
+    @Test
+    public void testOracleScriptWithBeginEndBlock() {
+        String sql = getClasspathSqlLocator().getResource("script/oracle-with-begin-end-blocks.sql");
+        try (Script script = new Script(HandleAccess.createHandle(), sql)) {
+
+            List<String> statements = script.getStatements();
+
+            assertThat(statements).hasSize(3);
+
+            String lastStmt = statements.get(2);
+            assertThat(lastStmt).startsWith("CREATE OR REPLACE TRIGGER EXAMPLE_TRIGGER");
+            assertThat(lastStmt).endsWith("END;");
+            assertThat(lastStmt).hasLineCount(15);
+            assertThat(lastStmt).has(new Condition<>(s -> 7 == s.chars().filter(ch -> ch == ';').count(), "count semicolons"));
+
+        } catch (SQLException ex) {
```

Comment:
You could just make the test `throws SQLException` which will automatically fail the test.

Reply:
Ok, will do